demisto_sdk/dev_tools/initiator.py

```diff
@@ -0,0 +1,286 @@
+            fp.write('[]')
+
+        fp = open(self.full_path + '/.secrets-ignore', 'a')
+        fp.close()
+
+        fp = open(self.full_path + '/.pack-ignore', 'a')
+        fp.close()
+
+        print_color(f"Successfully created the pack {self.name} in: {self.full_path}", LOG_COLORS.GREEN)
+
+        print("\nDo you want to create an integration in the pack? Y/N")
+        create_integration = str(input()).lower()
+        if create_integration == 'y':
+            self.check_name(created_object="integration")
+            self.output_dir = self.full_path + INTEGRATIONS_DIR
+            return self.integration_init()
+
+        return True
+
+    def integration_init(self) -> bool:
+        """Creates a new integration according to a template.
+
+        Returns:
+            bool. True if the integration was created successfully, False otherwise.
+        """
+        # if output directory given create the integration there
+        if len(self.output_dir) > 0:
+            self.full_path = self.output_dir + '/' + self.name
+
+        # the file pack-metadata indicates we are in a pack directory
+        # thus we will create the integration under the Integrations directory of the pack
+        elif os.path.isfile("pack-metadata.json"):
+            self.full_path = 'Integrations/' + self.name
+
+        # if non of the conditions above apply - create the integration in the local directory
+        else:
+            self.full_path = self.name
+
+        if not self.create_new_directory():
+            return False
+
+        hello_world_path = os.path.normpath(os.path.join(__file__, "..", "..", 'common', 'templates',
+                                                         HELLO_WORLD_INTEGRATION))
+
+        copy_tree(str(hello_world_path), self.full_path)
+        if self.name != HELLO_WORLD_INTEGRATION:
+            self.rename(current_suffix=HELLO_WORLD_INTEGRATION)
+            self.yml_reformatting(current_suffix=HELLO_WORLD_INTEGRATION)
+
+        print_color(f"Finished creating integration: {self.full_path}.", LOG_COLORS.GREEN)
+
+        return True
+
+    def script_init(self) -> bool:
+        """Creates a new script according to a template.
+
+        Returns:
+            bool. True if the script was created successfully, False otherwise.
+        """
+        # if output directory given create the script there
+        if len(self.output_dir) > 0:
+            full_path = self.output_dir + '/' + self.name
+
+        # the file pack-metadata indicates we are in a pack directory
+        # thus we will create the script under the Scripts directory of the pack
+        elif os.path.isfile("pack-metadata.json"):
+            full_path = 'Scripts/' + self.name
+
+        # if non of the conditions above apply - create the integration in the local directory
+        else:
+            full_path = self.name
+
+        if not self.create_new_directory():
+            return False
```

Comment:
those lines are quite repetitive, maybe we should make a function out of it?

Reply:
I though the separation is good since it creates two distinct flows for scripts and integrations - your call